osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -333,21 +333,21 @@ public double ComputeScore(ScoringMode mode, ScoringValues current, ScoringValue
         /// <param name="totalBasicHitObjects">The total number of basic (non-tick and non-bonus) hitobjects in the beatmap.</param>
         /// <returns>The total score computed from the given scoring component ratios.</returns>
         [Pure]
-        public double ComputeScore(ScoringMode mode, double accuracyRatio, double comboRatio, double bonusScore, int totalBasicHitObjects)
+        public long ComputeScore(ScoringMode mode, double accuracyRatio, double comboRatio, long bonusScore, int totalBasicHitObjects)
         {
             switch (mode)
             {
                 default:
                 case ScoringMode.Standardised:
                     double accuracyScore = accuracyPortion * accuracyRatio;
                     double comboScore = comboPortion * comboRatio;
-                    return (max_score * (accuracyScore + comboScore) + bonusScore) * scoreMultiplier;
+                    return (long)((max_score * (accuracyScore + comboScore) + bonusScore) * scoreMultiplier);
```

Comment:
Fairly certain this shouldn't be rounded when called recursively via the `Classic` path.